<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="3" passed="2" failed="1" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2023-07-19T14:34:18 GMT" name="TestSuite2" finished-at="2023-07-19T14:34:26 GMT" duration-ms="7534">
    <groups>
    </groups>
    <test started-at="2023-07-19T14:34:18 GMT" name="CreateUserTest" finished-at="2023-07-19T14:34:24 GMT" duration-ms="6397">
      <class name="chaining.CreateUser">
        <test-method signature="createUser(org.testng.ITestContext)[pri:0, instance:chaining.CreateUser@4de4b452]" started-at="2023-07-19T14:34:18 GMT" name="createUser" finished-at="2023-07-19T14:34:24 GMT" duration-ms="6269" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[org.testng.TestRunner@8458f04]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- createUser -->
      </class> <!-- chaining.CreateUser -->
    </test> <!-- CreateUserTest -->
    <test started-at="2023-07-19T14:34:24 GMT" name="GetUserTest" finished-at="2023-07-19T14:34:26 GMT" duration-ms="1019">
      <class name="chaining.GetUser">
        <test-method signature="getAllUsers()[pri:0, instance:chaining.GetUser@209da20d]" started-at="2023-07-19T14:34:25 GMT" name="getAllUsers" finished-at="2023-07-19T14:34:26 GMT" duration-ms="1004" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getAllUsers -->
        <test-method signature="getUserById(org.testng.ITestContext)[pri:0, instance:chaining.GetUser@209da20d]" started-at="2023-07-19T14:34:26 GMT" name="getUserById" finished-at="2023-07-19T14:34:26 GMT" duration-ms="1" status="FAIL">
          <params>
            <param index="0">
              <value>
                <![CDATA[org.testng.TestRunner@22002459]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "java.lang.Integer.intValue()" because the return value of "org.testng.ITestContext.getAttribute(String)" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "org.testng.ITestContext.getAttribute(String)" is null
at chaining.GetUser.getUserById(GetUser.java:29)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:564)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:829)
at org.testng.TestRunner.run(TestRunner.java:602)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
at org.testng.SuiteRunner.run(SuiteRunner.java:330)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
at org.testng.TestNG.runSuites(TestNG.java:1099)
at org.testng.TestNG.run(TestNG.java:1067)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getUserById -->
      </class> <!-- chaining.GetUser -->
    </test> <!-- GetUserTest -->
  </suite> <!-- TestSuite2 -->
</testng-results>
